internal/persistence/sql/namespaces.go

```diff
@@ -0,0 +1,87 @@
+package sql
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/ory/keto/internal/namespace"
+)
+
+type namespaceRecord struct {
+	NetworkID uuid.UUID `db:"nid"`
+	ID        uuid.UUID `db:"id"`
+	SerialID  int       `db:"serial_id"`
+	CreatedAt time.Time `db:"created_at"`
+	UpdatedAt time.Time `db:"updated_at"`
+}
+
+func (*namespaceRecord) TableName(context.Context) string {
+	return "keto_namespace_ids"
+}
+
+func (p *Persister) cacheNamespaceMapping(serialID int, id uuid.UUID) {
+	p.namespaceIDCache.Set(serialID, id, 1)
+	p.namespaceIDCache.Set(id[:], serialID, 1)
```

Comment:
Either prefix the caches with e.g. `serial-${id}`, `${pk}-id` or use two separate caches to avoid accidentally confusing these values.

Reply:
Isn't this already handled by the different types the keys have?

Reply:
Yeah, but I would prefer to make it explicit. Cache key confusion is a security risk here.

Reply:
The cache is currently shared by all networks so need to fix that in the first place 😅